pilot/pkg/bootstrap/server.go

```diff
@@ -840,13 +841,19 @@ func (s *Server) cachesSynced() bool {
 func (s *Server) initRegistryEventHandlers() {
 	log.Info("initializing registry event handlers")
 	// Flush cached discovery responses whenever services configuration change.
-	serviceHandler := func(svc *model.Service, _ model.Event) {
-		pushReq := &model.PushRequest{
-			Full:           true,
-			ConfigsUpdated: sets.New(model.ConfigKey{Kind: kind.ServiceEntry, Name: string(svc.Hostname), Namespace: svc.Attributes.Namespace}),
-			Reason:         []model.TriggerReason{model.ServiceUpdate},
+	serviceHandler := func(prev, curr *model.Service, event model.Event) {
```

Comment:
This is the main change

Reply:
For any change really: we MUST have a setting, ideally off by default if it is risky.